### ISSUE TITLE
Nested List_

### ISSUE THREAD

**Issue by ondrejmirtes** [nikic/PHP-Parser#321]
First of all, I really want to thank you for continuing working on PHP-Parser! It's been a really useful tool for several of my projects.

I just tried out 3.0.0 and found probably a bug with new handling of List_. Take this example (https://3v4l.org/8TeXR):

```
$someArray = ['foo', 'bar', 'ipsum', ['baz', 'dolor']];
list($listedOne, , $listedTwo['two'], list($listedThree, $listedFour['four'])) = $someArray;
```

There's a nested list() calls. I'd expect the first top list() to contain array of ArrayItems, but mine contains in this case:

```
ArrayItem
NULL (expected, but not said in the typehint of List_::$items)
ArrayItem
List_
```

I'd expect them to all be ArrayItem. In the case of the nested List_, I'd like to be an ArrayItem with List_ in its $value property.

**Comment by nikic** [comment]
Thanks for catching this so quickly! Fixed in https://github.com/nikic/PHP-Parser/commit/3e158a2313581ccedce961b8c73ec34cc1159ce4 and released as [v3.0.1](https://github.com/nikic/PHP-Parser/releases/tag/v3.0.1).

**Comment by ondrejmirtes** [comment]
@nikic By the way, this is the project I use PHP-Parser on: https://medium.com/@ondrejmirtes/phpstan-2939cd0ad0e3 (https://github.com/phpstan/phpstan) :)
